test/apps/iodemo/io_demo.cc

```diff
@@ -135,33 +141,52 @@ const unsigned IoDemoRandom::_M = 0x7fffffffU;
 class P2pDemoCommon : public UcxContext {
 protected:
 
-    /* IO request header */
-    typedef struct {
-        io_op_t     op;
+    /* data validation header */
+    typedef struct __attribute__ ((packed)) {
+        uint16_t    chk_sum;
+    } chk_hdr_t;
+
+    /* transaction header */
+    typedef struct __attribute__ ((packed)) {
         uint32_t    sn;
+    } tr_hdr_t;
```

Comment:
seems tr_hdr_t and chk_hdr_t always used together, can we unite them to 1 struct?

Reply:
the idea was to include sn to chk_sum, but since we check it separately, it seems redundant, WDYT?

Reply:
not sure i understand the question correctly, but sn could be part of the inputs which influence chk_sum, my comment was just that we can unite them to 1 struct to reduce some of the code

Reply:
per offline discussion - better don't start crc from the middle of header